File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.io.IOException;
import java.util.*;

import com.pinyougou.entity.PageResult;
import com.pinyougou.entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        List<TbBrand> all = brandService.findAll();
        return all;
    }

    @RequestMapping("/findPage")
    public PageResult findPage(@RequestBody TbBrand brand,@RequestParam("pageNum") int pageNum, @RequestParam("pageSize") int pageSize) {
        return brandService.findPage(brand,pageNum, pageSize);
    }

    @RequestMapping("/insertBrand")
    public Result insertBrand(@RequestBody TbBrand brand)
    {
        try {

            brandService.addBrand(brand);
            return new Result(true,"add brand success");

        }catch (Exception e)
        {
            return new Result(false,"add brand failure");
        }
    }

    @RequestMapping("/modifyBrand")
    public Result modifyBrand(@RequestBody TbBrand brand)
    {
        try {
            brandService.modifyBrand(brand);
            return new Result(true,"success");
        }catch (Exception e){
            return new Result(false,"modify brand failure");
        }
    }

    @RequestMapping("/findBrandById")
    public Result findOne(String id)
    {
        try {

            TbBrand brandById = brandService.findBrandById(id);
            return new Result(true,"success", brandById);
        }catch (Exception e){
            return new Result(false,"failure",null);
        }

    }

    @RequestMapping("/batchDel")
    public Result batchDel(String[] ids){
        try {
            brandService.batchDel(ids);
            return new Result(true,"success",null);
        }catch (Exception e){
            return new Result(false,"failure",null);
        }
    }

    @RequestMapping("/getBrandOptions")
    public List<Map<String,String>> getOptions(){

        List<Map<String, String>> brandOptions = brandService.getBrandOptions();

        return brandOptions;

    }
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController',function ($scope,$controller,brandService){

    $controller('baseController',{$scope:$scope});

        //查询品牌列表
        $scope.findAll=function(){
            brandService.findAll().success(
                function(response){
                    $scope.list=response;
                }
            );
        }

        //搜索
        $scope.searchEntity={};
        $scope.search=function (pageNum,pageSize) {

            brandService.search(pageNum,pageSize,$scope.searchEntity).success(
                function (response) {
                    $scope.paginationConfig.totalItems=response.total;
                    $scope.list=response.rows;
                }
            );

        }

        //分页查询
        $scope.findPage=function (pageNum,pageSize) {

            brandService.findPage().success(

                function (response) {

                    $scope.list=response.rows;

                    $scope.paginationConfig.totalItems=response.total;
                }
            );
        }

        //添加更新品牌
        $scope.addBrand=function () {

            if ($scope.entity.id != null){
                //更新
                brandService.updateBrand($scope.entity).success(
                    function (response) {

                        if (response.success){

                            $scope.reloadList();

                        }else {
                            alert("update brand failure")
                        }


                    }
                );
            }else {
                //添加
                brandService.addBrand($scope.entity).success(
                    function (response) {

                        if(response.success)
                        {
                            $scope.reloadList();
                        }else {
                            alert(response.message);
                        }

                    }
                )
            }

        };

        //查询品牌
        $scope.findOne=function (id) {

            brandService.findOne(id).success(
                function (response) {

                    if(response.success){

                        $scope.entity=response.brand;
                    }
                }
            );
        }


        //批量删除
        $scope.batchDel=function () {

            brandService.batchDel($scope.selectedIds).success(
                function (response) {

                    if (response.success){
                        $scope.reloadList();
                    }else {
                        alert("delete failure");
                    }

                }
            );

        }



    }
    
);